### ISSUE TITLE
Feature Proposal: no [OverrideBuiltins]

### ISSUE THREAD

**Issue by domenic** [WICG/document-policy#16]
See https://github.com/whatwg/html/issues/4112. The case people are most concerned about are when you could introduce an element with a name or ID that overrides the built-ins, i.e. the [OverrideBuiltins] cases on Document and HTMLFormElement.

You could imagine various expansions of this, e.g.

- Turning off named getters/setters entirely. (Although I'm not sure we have replacements in 100% of cases)
- Turning off "bad legacy DOM and HTML APIs" entirely. (E.g. things like Attr nodes.)

**Comment by petamoriken** [comment]
What is the current status of this proposal? I believe it would be effective to stop DOM clobbering against `window`, `document`, and `HTMLFormElement` using `Document-Policy: override-builtins=?0`.

**Comment by domfarolino** [comment]
I don't think it would help for clobbering against `window`, given https://html.spec.whatwg.org/#named-access-on-the-window-object:named-properties-object and https://webidl.spec.whatwg.org/#dfn-named-properties-object (in contrast with https://webidl.spec.whatwg.org/#legacy-platform-object-ownpropertykeys). Your demonstration in https://github.com/whatwg/html/issues/4112#issue-372852104 also seems to show this.

> What is the current status of this proposal? 

I think there is no status; it's probably waiting for someone to get together a short spec and drive it.

**Comment by annevk** [comment]
I think if we did something here we should definitely try to solve for `window` as well. That's the most egregious case.
